src/pybind/mgr/snap_schedule/fs/schedule_client.py

```diff
@@ -103,9 +103,22 @@ class SnapSchedClient(CephfsClient):
 
     def __init__(self, mgr):
         super(SnapSchedClient, self).__init__(mgr)
-        # TODO maybe iterate over all fs instance in fsmap and load snap dbs?
+        # Each db connection is now guarded by a Lock; this is required to
+        # avoid concurrent DB transactions when more than one paths in a
+        # file-system are scheduled at the same interval eg. 1h; without the
+        # lock, there are races to use the same connection, causing  nested
+        # transactions to be aborted
         self.sqlite_connections = {}
         self.active_timers = {}
+        self.conn_lock = Lock()  # lock to protect add/lookup db connections
```

Comment:
Not seen this `Lock` added in original PR.
Is that part of another backport?

Reply:
@nmshelke Here's the link to [line 154 in the original comit on master branch](https://github.com/ceph/ceph/blob/57c2ae079b42cde0e8f3deedc760183af81503b2/src/pybind/mgr/snap_schedule/fs/schedule_client.py#L154)

master branch and pacific have diverged significantly and so this delta that doesn't match with the master branch commit

Reply:
So maybe we should list this difference in the commit comments ?

Reply:
actually this PR is supposed to go in after #45189 and hence the diff